pkg/controllers/namespacesync/controller.go

```diff
@@ -52,12 +52,23 @@ func (r *Reconciler) SetupWithManager(
 						if !ok {
 							return false
 						}
-						return r.TargetNamespaceFilter(ns)
+						return r.IsTargetNamespace(ns)
 					},
 					UpdateFunc: func(e event.UpdateEvent) bool {
 						// Called when an object is already in the cache, and it is either updated,
 						// or fetched as part of a re-list (aka re-sync).
-						return false
+						nsOld, ok := e.ObjectOld.(*corev1.Namespace)
+						if !ok {
+							return false
+						}
+						nsNew, ok := e.ObjectNew.(*corev1.Namespace)
+						if !ok {
+							return false
+						}
+						// Only reconcile the namespace if the answer to the question "Is this a
+						// target namespace?" has changed. Other changes are not relevant to
+						// this controller.
+						return r.IsTargetNamespace(nsOld) != r.IsTargetNamespace(nsNew)
```

Comment:
Do we also want to skip the case where the user removed the label from the namespace?
```suggestion
						return r.IsTargetNamespace(nsNew) && r.IsTargetNamespace(nsOld) != r.IsTargetNamespace(nsNew)
```

Reply:
Good observation! We can skip every case except the one where the label is added. Done in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/775/commits/6553aff0210d8db0da37ca50437dda09de5f4399